cmd/update.go

```diff
@@ -68,15 +70,36 @@ func gup(cmd *cobra.Command, args []string) int {
 		print.Err(err)
 		return 1
 	}
+
+	exclude, err := cmd.Flags().GetString("exclude")
+	if err != nil {
+		print.Err(fmt.Errorf("%s: %w", "can not parse command line argument (--exclude)", err))
+		return 1
+	}
+
 	pkgs = extractUserSpecifyPkg(pkgs, args)
 
+	pkgs = excludePkgs(exclude, pkgs)
+
 	if len(pkgs) == 0 {
 		print.Err("unable to update package: no package information")
 		return 1
 	}
 	return update(pkgs, dryRun, notify, cpus)
 }
 
+func excludePkgs(exclude string, pkgs []goutil.Package) []goutil.Package {
+	excludelist := strings.Split(exclude, ",")
+
+	for i := len(pkgs) - 1; i >= 0; i-- {
+		if slice.Contains(excludelist, pkgs[i].Name) {
+			pkgs = append(pkgs[:i], pkgs[i+1:]...)
+		}
+	}
+
+	return pkgs
+}
```

Comment:
I would like to make a small modification. The code you wrote uses less memory. However, The following writing style is simpler and easier to read.

```suggestion
func excludePkgs(exclude string, pkgs []goutil.Package) []goutil.Package {
	excludeList := strings.Split(exclude, ",")

	packageList := []goutil.Package{}
	for _, v := range pkgs {
		if slice.Contains(excludeList, v.Name) {
			continue
		}
		packageList = append(packageList, v)
	}

	return pkgs
}
```